core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -72,17 +100,202 @@ private FunctionResolver date() {
         impl(nullMissingHandling(DateTimeFunction::exprDate), DATE, TIMESTAMP));
   }
 
+  /**
+   * Specify a start date and add a temporal amount to the date.
+   * The return type depends on the date type and the interval unit. Detailed supported signatures:
+   * (DATE, DATETIME/TIMESTAMP, INTERVAL) -> DATETIME
+   * (DATE, LONG) -> DATE
+   * (DATETIME/TIMESTAMP, LONG) -> DATETIME
+   */
+  private FunctionResolver date_add() {
+    return define(BuiltinFunctionName.DATE_ADD.getName(),
+        impl(nullMissingHandling(DateTimeFunction::exprAddDateInterval), DATE, DATE, INTERVAL),
+        impl(nullMissingHandling(DateTimeFunction::exprAddDateInterval), DATETIME, DATE, INTERVAL),
```

Comment:
why the function has same input signature but has different output signature?

Reply:
some use cases required different return type. E.g `select date_add(date('2020-09-16'), interval 1 hour)` should return `2020-09-16 01:00:00` but `select date_add(date('2020-09-16'), interval 1 day)` should return `2020-09-17`.  But this was always returning datetime in schema  anyway so removed it.